tests/service/SyntaxTreeTests/SynTypeTests.fs

```diff
@@ -49,3 +49,65 @@ type T =
       ])) ->
         assertRange (3, 14) (3, 56) mTuple
     | _ -> Assert.Fail $"Could not get valid AST, got {parseResults}"
+
+[<Test>]
+let ``SynType.Or inside SynExpr.TraitCall`` () =
+    let parseResults =
+        getParseResults
+             """
+let inline (!!) (x: ^a) : ^b = ((^a or ^b): (static member op_Implicit: ^a -> ^b) x)
```

Comment:
Some nice examples are on the bottom of this docs:
https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/generics/statically-resolved-type-parameters 

Especially the combination of having **or** and **and** clauses together.